js/passwordManager/passwordManager.js

```diff
@@ -0,0 +1,219 @@
+// Bitwarden password manager. Requires session key to unlock the vault.
+class Bitwarden {
+  constructor() {
+    this.sessionKey = null
+    this.lastCall = null
+    this.name = 'Bitwarden'
+  }
+
+  // Checks if given command is a valid Bitwarden-CLI command.
+  async _checkCommand(command) {
+    try {
+      let process = new ProcessSpawner(command, ['--version'])
+      let data = await process.execute()
+      if (data.length > 0) {
+        return true
+      }
+    } catch (ex) {
+      return false
+    }
+  }
+
+  // Returns a Bitwarden-CLI tool path by checking possible locations.
+  // First it checks if the tool was installed for Min specifically by
+  // by checking the settings value. If that is not set or doesn't point
+  // to a valid executable, it check the if 'bw' is available globally.
+  async _getToolPath() {
+    let localPath = settings.get('bitwardenPath')
+    if (localPath) {
+      let local = await this._checkCommand(localPath)
+      if (local) {
+        return localPath
+      }
+    }
+
+    let global = await this._checkCommand('bw')
+    if (global) {
+      return 'bw'
+    }
+
+    return null
+  }
+
+  // Checks if Bitwarden integration is configured properly by trying to
+  // obtain a valid Bitwarden-CLI tool path.
+  async checkIfConfigured() {
+    this.path = await this._getToolPath()
+    if (this.path == null) {
+      return false
+    }
+
+    isCommandValid = await this._checkCommand(this.path)
```

Comment:
I'm not sure I understand why this is necessary; _getToolPath already verifies the path before it returns it, so it shouldn't be possible for this to fail. Is there a situation where that could happen?

Reply:
I just forgot that _getToolPath alredy does that. >.< Reverted